src/main/java/com/amazon/ion/impl/macro/MacroEvaluatorAsIonReader.kt

```diff
@@ -148,10 +148,30 @@ class MacroEvaluatorAsIonReader(
 
     override fun getTypeAnnotations(): Array<String>? = currentValueExpression?.annotations?.let { Array(it.size) { i -> it[i].assumeText() } }
     override fun getTypeAnnotationSymbols(): Array<SymbolToken>? = currentValueExpression?.annotations?.toTypedArray()
-    // TODO: Make this into an iterator that unwraps the SymbolTokens as it goes instead of allocating a new list
+
+    private class SymbolTokenAsStringIterator(val tokens: List<SymbolToken>) : MutableIterator<String> {
+
+        var index = 0
+
+        override fun hasNext(): Boolean {
+            return index < tokens.size
+        }
+
+        override fun next(): String {
+            return tokens[index++].assumeText()
+        }
```

Comment:
This ought to throw `NoSuchElementException` when `index >= tokens.size` (see [Iterator.next](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/-iterator/next.html)), but this will instead throw [ArrayIndexOutOfBoundsException](https://pl.kotl.in/72XdyoQ0O).